crates/red_knot_python_semantic/src/types.rs

```diff
@@ -356,6 +407,34 @@ impl<'db> Type<'db> {
             _ => Type::Unknown, // TODO type errors
         }
     }
+
+    /// Given a type that is assumed to represent an instance of a class,
+    /// return a type that represents that class itself.
+    #[must_use]
+    pub fn to_type_of_class(&self, db: &'db dyn Db) -> Type<'db> {
```

Comment:
Nit: `to_class_type`. Although I can see how this is slightly confusing with `Type::class`. But I think that also applies to `to_type_of_class`.

Reply:
Very hard to find a good name here... I initially had `to_type()`, but then I realised that would end up as `Type::to_type`, which is awful 😆

Reply:
I prefer my existing name here, I think it's marginally less confusing. But neither is great and I don't have a very strong opinion.

Reply:
I prefer `to_class_type`; I'm unable to detect any clarity improvement in `to_type_of_class`. An alternative would be `to_meta_type`, but I don't think that's better than `to_class_type`.

Reply:
Oh I actually like `to_meta_type` quite a lot! I think it's much clearer that you might not end up with a `Type::Class` variant or `ClassType` instance at the end of it. Mind if I go with that?

Reply:
No, I have no problem going with that!